plugins/PRAplugin/src/graphModel.py

```diff
@@ -0,0 +1,238 @@
+          nodes.pop(mapping[key],None)
+          for node in nodes.keys():
+            if mapping[key] in nodes[node]:
+              nodes[node].remove(mapping[key])
+
+    ravenGraph = GS.graphObject(nodes)
+
+    dictOut = {}
+    for nodeO in container.nodesOUT:
+      paths = []
+      for nodeI in container.nodesIN:
+        paths = paths + ravenGraph.findAllPaths(nodeI,nodeO)
+      var = container.InvMapping[nodeO]
+      if paths:
+        dictOut[var] = np.asarray(0.)
+      else:
+        dictOut[var] = np.asarray(1.)
+    return dictOut
+
+  def runTimeDep(self, container, Inputs):
+    """
+      This method performs a dynamic analysis of the graph model
+      @ In, container, object, self-like object where all the variables can be stored
+      @ In, Inputs, dict, dictionary of inputs from RAVEN
+      @ Out, outcome, dict, dictionary containing the temporal status of all output variables
+    """
+    times = []
+    times.append(0.)
+    for key in Inputs.keys():
+      if key in container.mapping.keys() and Inputs[key][0]!=1.:
+        times.append(Inputs[key][0])
+    times = sorted(times, key=float)
+
+    outcome={}
+    for var in container.nodesOUT:
+      outcome[container.InvMapping[var]] = np.asarray([0.])
+
+    for time in times:
+      inputToPass=self.inputToBePassed(container,time,Inputs)
+      tempOut = self.runStatic(container, inputToPass)
+      for var in tempOut.keys():
+        if tempOut[var] == 1.:
+          if time == 0.:
+            outcome[var] = np.asarray([1.])
+          else:
+            if outcome[var][0] > 0:
+              pass
+            else:
+              outcome[var] = np.asarray([time])
+    return outcome
+
+  def inputToBePassed(self,container,time,Inputs):
+    """
+      This method returns the status of the input variables at time t=time
+      @ In, container, object, self-like object where all the variables can be stored
+      @ In, Inputs, dict, dictionary of inputs from RAVEN
+      @ In, time, float, time at which the input variables need to be evaluated
+    """
+    inputToBePassed = {}
+    for key in Inputs.keys():
+      if key in container.mapping.keys():
+        if Inputs[key][0] == 0. or Inputs[key][0] == 1.:
+          inputToBePassed[key] = Inputs[key]
+        else:
+          if Inputs[key][0] > time:
+            inputToBePassed[key] = np.asarray([0.])
+          else:
+            inputToBePassed[key] = np.asarray([1.])
+    return inputToBePassed
+
+def findAllRecursive(node, element):
+  """
+    A function for recursively traversing a node in an elementTree to find
+    all instances of a tag.
+    Note that this method differs from findall() since it goes for all nodes,
+    subnodes, subsubnodes etc. recursively
+    @ In, node, ET.Element, the current node to search under
+    @ In, element, str, the string name of the tags to locate
+    @ InOut, result, list, a list of the currently recovered results
+  """
+  result=[]
```

Comment:
this method is redundant... it is the 3rd time I see this... can you add in utils?